### PR TITLE
Misc pod fixes

### PR DESCRIPTION
**Pull request by mheon**
Fix a segfault and an issue where we were deleting the pod's /etc/hosts

### CODE REVIEW

File: libpod/runtime_pod_infra_linux.go [hunk 1]
```diff
@@ -31,17 +31,24 @@ func (r *Runtime) makeInfraContainer(ctx context.Context, p *Pod, imgName, imgID
 	isRootless := rootless.IsRootless()
 
 	entryCmd := []string{r.config.InfraCommand}
-	// default to entrypoint in image if there is one
-	if len(config.Entrypoint) > 0 {
-		entryCmd = config.Entrypoint
+	if config == nil {
```
Comment (haircommander) [inline libpod/runtime_pod_infra_linux.go]: Did you mean to leave this?

**Comment by haircommander** [comment]
ah beat me to both fixes. changes LGTM

**Comment by giuseppe** [comment]
I was carrying the segfault patch in a different PR as tests were failing on it.  I'll rebase on top of this.

/lgtm

**Comment by mheon** [comment]
I don't think the test failures are me - missing registries.conf in the test env?

**Comment by mheon** [comment]
Nevermind, found the problem - related to how we generated the command for infra containers.

**Comment by giuseppe** [comment]
LGTM

**Comment by giuseppe** [comment]
this PR is to fix a regression in the tests and it is blocking other PRs:

/lgtm

**Pull request merged**